web/lib/supabase/contracts.ts

```diff
@@ -111,8 +111,15 @@ export const getPublicContracts = async (options: {
   limit: number
   beforeTime?: number
   order?: 'asc' | 'desc'
+  ids?: string[]
```

Comment:
I think this should be a separate function. The other params would generally not be used when getting all by id.